mlproc.py

```diff
@@ -151,10 +108,51 @@ def _start(self):
         e = mp.Event()
 
         p = mp.Process(
-            target=_start_matlab,
+            target=self._start_matlab,
             args=(e, self._alive_check_period,
                   self._startup_options, self._shared_name))
 
         p.daemon = False
         p.start()
         e.wait()
+
+    # --------------------------------------------------------------------------
+    def _start_matlab(self, event: mp.Event,
+                    alive_check_period=60, startup_options=None, shared_name=None):
+        
+        pid = mp.current_process().pid
+        self._pid.value = pid
+
+        print('Start matlab process %s with process ID {}'.format(pid) % shared_name )
+
+        event.clear()
+
+        if startup_options is None:
+            eng = me.start_matlab()
+        else:
+            eng = me.start_matlab(startup_options)
+
+        try:
+            if shared_name:
+                eng.matlab.engine.shareEngine(shared_name, nargout=0)
+            else:
+                eng.matlab.engine.shareEngine(nargout=0)
+                shared_name = eng.matlab.engine.engineName(nargout=1)
+
+            print('Matlab shared engine {} is started'.format(shared_name))
+        except me.MatlabExecutionError:
+            raise
+        finally:
+            event.set()
+
+        while self._pid.value != -1:
+            # Is matlab alive?
+            try:
+                eng.version(nargout=0)
+            except me.EngineError:
+                break
+            time.sleep(alive_check_period)
+
+        if eng._check_matlab(): eng.exit()
```

Comment:
Is this code works for all supported MATLAB versions and all supported platforms?
In fact, I do not remember why we made that crutch here before, but previous code looked just awfully. :)

Reply:
I am afraid I cannot test on Mac. However, `_check_matlab()` and `exit()` are standard methods of `MatlabEngine`.